core/math/math_funcs.h

```diff
@@ -33,14 +33,14 @@
 
 #include "core/error/error_macros.h"
 #include "core/math/math_defs.h"
-#include "core/math/random_pcg.h"
```

Comment:
This *also* breaks on windows, which seems to mean that this code needs to be moved to somewhere else:
```cpp
#if defined(__GNUC__)
#define CLZ32(x) __builtin_clz(x)
#elif defined(_MSC_VER)
#include <intrin.h>
static int __bsr_clz32(uint32_t x) {
	unsigned long index;
	_BitScanReverse(&index, x);
	return 31 - index;
}
#define CLZ32(x) __bsr_clz32(x)
#else
#endif
```

Reply:
Well, then, should this solve the errors on Windows?
```c++
#include <float.h>
#ifdef _MSC_VER
#include <intrin.h>
#endif
```

Reply:
It does, but the `float` change removes the main reason here, save for the random part, which isn't very heavy (the third-party code is tiny)